app/Bots/IRCAnnounceBotExternal.php

```diff
@@ -0,0 +1,175 @@
+ * @license    https://www.gnu.org/licenses/agpl-3.0.en.html/ GNU Affero General Public License v3.0
+ */
+
+namespace App\Bots;
+
+use App\Models\TmdbMovie;
+use App\Models\TmdbTv;
+use App\Models\Torrent;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Facades\Log;
+use Throwable;
+
+class IRCAnnounceBotExternal
+{
+    public static function postAnnounceMsg(Torrent $torrent): bool
+    {
+        if (! config('irc-bot-external.is_enabled')) {
+            return false;
+        }
+
+        $appurl = config('app.url');
+        $announceTypeEnum = 0; // 0 NEW
+
+        $originEnum = match (true) {
+            $torrent->personal_release == true => 2,
+            $torrent->internal                 => 1,
+            default                            => 0,
+        };
+
+        $leechTypeEnum = match ($torrent->free) {
+            100     => 1,
+            75      => 2,
+            50      => 3,
+            25      => 4,
+            default => 0,
+        };
+
+        if ($torrent->doubleup) {
+            $leechTypeEnum = match ($leechTypeEnum) {
+                0 => 5,
+                1 => 6,
+                2 => 7,
+                3 => 8,
+                4 => 9,
+            };
+        }
+
+        $meta = null;
+        $category = $torrent->category;
+
+        if ($torrent->tmdb_movie_id > 0 || $torrent->tmdb_tv_id > 0) {
+            $meta = match (true) {
+                $category->tv_meta    => TmdbTv::find($torrent->tmdb_tv_id),
+                $category->movie_meta => TmdbMovie::find($torrent->tmdb_movie_id),
+                default               => null,
+            };
+        }
+
+        return self::post([
+            'id'                     => $torrent->id,
+            'url'                    => \sprintf('%s/torrents/%d', $appurl, $torrent->id),
+            'name'                   => $torrent->name,
+            'uploader'               => $torrent->anon ? 'Anonymous' : $torrent->user->username,
+            'size'                   => $torrent->getSize(),
+            'size_bytes'             => $torrent->size,
+            'announce_type_enum'     => $announceTypeEnum,
+            'category_enum'          => $category->id,
+            'category_name'          => $category->name,
+            'origin_enum'            => $originEnum,
+            'leech_type_enum'        => $leechTypeEnum,
+            'upload_time_unix_epoch' => $torrent->created_at->getTimestamp(),
+            'freeleech'              => (bool) $torrent->free > 0,
+            'freeleech_percent'      => $torrent->free,
+            'double_up'              => $torrent->doubleup,
+            'resolution'             => $torrent->resolution?->name ?? '',
+            'type'                   => $torrent->type->name,
+            'release_year'           => isset($meta->release_date) ? $meta->release_date->format('Y') : (isset($meta->first_air_date) ? $meta->first_air_date->format('Y') : null),
```

Comment:
```suggestion
            'release_year'           => $meta?->release_date?->format('Y') ?? $meta?->first_air_date?->format('Y'),
```

Reply:
Or:

```suggestion
            'release_year'           => $meta?->release_date?->format('Y') ?? $meta?->first_air_date?->format('Y') ?? $meta?->first_release_date?->format('Y'),
```

If you want to include igdb video games as well (would also require adding a `$category->game_meta => IgdbMeta::find($torrent->igdb)` to the match above)